test/e2e/storage/testsuites/base.go

```diff
@@ -421,6 +427,64 @@ func convertTestConfig(in *PerTestConfig) volume.TestConfig {
 	}
 }
 
+// getSizeRangesIntersection takes two instances of storage size ranges and determines the
+// intersection of the intervals (if it exists) and return the minimum of the intersection
+// to be used as the claim size for the test.
+// if value not set, that means there's no minimum or maximum size limitation and we set default size for it.
+func getSizeRangesIntersection(first volume.SizeRange, second volume.SizeRange) (string, error) {
+	var firstMin, firstMax, secondMin, secondMax resource.Quantity
+	var err error
+	//if not set, use default size
+	if len(first.Min) == 0 {
+		first.Min = defaultMinSize
+	}
+	if len(first.Max) == 0 {
+		first.Max = defaultMaxSize
+	}
+	if len(second.Min) == 0 {
+		second.Min = defaultMinSize
+	}
+	if len(second.Max) == 0 {
+		second.Max = defaultMaxSize
+	}
+
+	if firstMin, err = resource.ParseQuantity(first.Min); err != nil {
+		ginkgo.By(fmt.Sprintf("Failed parsing quantity %s", first.Min))
```

Comment:
`ginkgo.By` is for logging steps in the test execution, not for logging of errors. But this is an utility function. It shouldn't log errors at all, but rather just return them and let the caller decide whether and how the error is to be reported.

You can remove all of these `ginkgo.By` lines.

Reply:
+1 here also.